pkg/operator/ceph/csi/config.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateUpdateClientProfileRadosNs(ctx context.Context, c client.Client, clusterInfo *cephclient.ClusterInfo, cephBlockPoolRadosNamespacedName types.NamespacedName, clusterID, clusterName string) error {
+
+	logger.Info("creating ceph-csi clientProfile cr for rados namespace")
+
+	csiOpClientProfile := &csiopv1a1.ClientProfile{}
+	csiOpClientProfile.Name = clusterID
+	csiOpClientProfile.Namespace = cephBlockPoolRadosNamespacedName.Namespace
+	csiOpClientProfile.Spec = csiopv1a1.ClientProfileSpec{
+		CephConnectionRef: v1.LocalObjectReference{
+			Name: clusterName,
+		},
+		Rbd: &csiopv1a1.RbdConfigSpec{
+			RadosNamespace: cephBlockPoolRadosNamespacedName.Name,
+		},
+	}
+
+	err := clusterInfo.OwnerInfo.SetOwnerReference(csiOpClientProfile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set owner reference on ClientProfile CR %q", csiOpClientProfile.Name)
+	}
+
+	err = c.Create(ctx, csiOpClientProfile)
```

Comment:
This looks like creating multiple `clientProfile` resources.  One for RadosNamespace and one for SubvolumeGroup (in `CreateUpdateClientProfileSubVolGrp` method below). 

Is it expected to have multiple instances of `clientProfile` CRD?

@Madhu-1 @subhamkrai

Reply:
yes we need to have one default clientProfile with cephcluster Namespace like we have one for `rook-ceph` or openshift-storage and also we need to create clientProfile per subvolumegroup and radosnamespace.

@subhamkrai i think creating default one is missing can you please check

Reply:
this should be good now